src/sdp/SDP.ts

```diff
@@ -375,15 +377,22 @@ export default class SDP {
    * negotiate with the back end to determine whether to use layers allocation header extension
    * this will not add the packet overhead unless negotiated to avoid waste
    */
-  withVideoLayersAllocationRtpHeaderExtension(): SDP {
-    const sections = SDP.splitSections(this.sdp);
+  withVideoLayersAllocationRtpHeaderExtension(previousSdp: SDP): SDP {
+    const url = `http://www.webrtc.org/experiments/rtp-hdrext/video-layers-allocation00`
 
+    // According to https://webrtc.googlesource.com/src/+/b62ee8ce94e5f10e0a94d6f112e715cc4d0cd9dc,
+    // RTP header extension ID change would result in a hard failure. Therefore if the extension exists
+    // in the previous SDP, use the same extension ID to avoid the failure. Otherwise use a new ID
+    let previousId = previousSdp ? previousSdp.getRtpHeaderExtensionId(url) : -1;
+    const id = previousId === -1 ? this.getUniqueRtpHeaderExtensionId(SDP.splitLines(this.sdp)) : previousId;
```

Comment:
To make this super safe, can we check that this ID is still unique and just not add it if so?

Reply:
Not sure if I fully understood this comment. Do you mean checking if `previousId` exists, checking for if it is unique in the current SDP?

Reply:
like if the first SDP uses 1-14 so we use 15. But for some reason browser adds a different extension for 15. We just bail out and don't add it.

Not sure if you saw in the previous PR, but we actually confirm whether the extension is negotiated before enabled any behavior based off of it. So bailing out and not adding is totally fine.

Reply:
I opened another PR https://github.com/aws/amazon-chime-sdk-js/pull/2195 for this. The change https://github.com/aws/amazon-chime-sdk-js/pull/2195/files#diff-d6a73856184ac9b6920494e5e8fcc8be715f515d30c8871cc9a8891ebc7c87fdR397 should avoid this behavior